package/src/custom/Extension/modules/Schedulers/Ext/ScheduledTasks/StudentGradebookJob.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+use Sugarcrm\Sugarcrm\custom\gradebook_fake\RecordManager;
+
+class StudentGradebookJob implements RunnableSchedulerJob
+{
+
+    /**
+     * @var SchedulersJob
+     */
+    protected $job;
+    /**
+     * @param SchedulersJob $job
+     */
+    public function setJob(SchedulersJob $job)
+    {
+        $this->job = $job;
+    }
+
+    /**
+     * This function defines the job that adds a new student to the GradebookFake app
+     * @param string $data The id for the new Student (contact) record
+     * @return bool true if a record was successfully created in the GradebookFake app
+     */
+    public function run($data)
+    {
+        if (!empty($data)) {
+            $bean = $this->getContactBean($data);
+
+            try {
+                //Call the external GradebookFake app to create a new record in it
+                $rm = $this->getRecordManager();
+                $success = $rm->createStudentRecord($bean->email1, $bean->first_name, $bean->last_name);
```

Comment:
`email1` shouldn't be used anymore. `SugarEmailAddress::getPrimaryAddress()` can be used to get the string of the primary email address. You don't have to change this now, but you'll want to create an issue to remind you to do it later.

Reply:
Since I was messing with the file anyway, I went ahead and fixed this.